src/pages/Home.tsx

```diff
@@ -1,44 +1,149 @@
+                })
+                //sets header accordingly so there is the right text at the top
+                setHeader("Nächste Vorlesungen");
+                return tomorrow;
+            } else {
+                setHeader("Morgige Vorlesungen");
+                return tomorrow;
+            }
+        } else {
+            setHeader("Heutige Vorlesung");
+            return today;
+        }
+
+    };
+    const [lectures, setLectures] = useState([] as Lecture[]); /// State for lectures that are shown
+    const [refreshing, setRefreshing] = useState(false); //for refreshing when reloading the site
+    const [header,setHeader] = useState("Heutige Vorlesungen"); //the header
+
+    const onRefresh = useCallback(() => {
+        // on refresh it shows the refresh button, then if it gets the right lecture it filters the list and then turns of the refresh
+        setRefreshing(true);
+        NetInfo.fetch().then(netstate=> {
+            if (netstate.isConnected) {
+                NetworkService.getLectures('MOS-TINF21A').then((lecture: Lecture[] | null) => {
+                    if (lecture) {
+                        setLectures(filterList(lecture))
+                        setRefreshing(false);
+                    }
+                })
+            } else {
+                let toast = Toast.show("Sie sind nicht zum Internet verbunden")
+                setRefreshing(false)
+            }
+        })
+
+    }, []);
+
+    //on startup it loads automatically the Data from async storatge so that it can be viewed in offline mode
+    useEffect(() => {
+        getData('lectures').then((lectureList) => {
+            if (lectureList) {
+                const lect: Lecture[] = lectureList.map((lecture: Lecture) => {
+                    const lec: Lecture = {
+                        id: lecture["id"],
+                        date: new Date(lecture["date"]),
+                        startTime: new Date(lecture["startTime"]),
+                        endTime: new Date(lecture["endTime"]),
+                        name: lecture["name"],
+                        type: lecture["type"],
+                        rooms: lecture["rooms"],
+                        course: lecture["course"]
+                    }
+                    return lec;
+                });
+                setLectures(filterList(lect));
+            }
+
+        });
+    }, []);
+
+    return (
+        <ScrollView style={scrollViewStyle.scrollView}
+                    refreshControl={
+                        <RefreshControl refreshing={refreshing}
+                                        onRefresh={onRefresh}
+                        />
+                    }
+        >
+            <Text style={style.header}>{header}</Text>
+
+            {
+                lectures.length == 0 ? <Text>Keine Vorlesungen in nächster Zeit</Text> : <></>
+            }
+            {lectures.map(lecture => {
+                return (<CalenderEntry name={lecture.name}
```

Comment:
Für cleaneren Code könntest du recherchieren ob man einfach `lecture` als objekt übergeben kann und der component die values aus dem objekt zieht

Reply:
```jsx
<CalenderEntry {…lecture}} />
```

Reply:
Geht am z.B. mit dem Spread Operator: https://reactjs.org/docs/jsx-in-depth.html#spread-attributes